src/fitness_tracker/ui.py

```
@@ -251,63 +257,106 @@ def apply_sensor_settings(
-        )
-        if not self.test_mode:
-            # Only spin BLE loops when not in test mode
-            self.recorder.start()
+                if current:
+                    try:
+                        if not current.shutdown():
+                            msg = "The current sensor worker is still shutting down; profile unchanged"
+                            logger.error(msg)
+                            self._run_on_ui_thread(lambda: self.show_toast(msg))
+                            return
+                    except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)